data_safe_haven/administration/users/guacamole_users.py

```diff
@@ -33,49 +29,18 @@ def __init__(self, config: Config, sre_name: str, *args: Any, **kwargs: Any):
             / "postgresql"
         )
         self.sre_name = sre_name
+        self.group_name = f"Data Safe Haven Users SRE {sre_name}"
 
     def add(self, users: Sequence[ResearchUser]) -> None:
         """Add sequence of users to Guacamole"""
-        # Update Guacamole users
-        users_to_add = []
-        for new_user in users:
-            if new_user in self.list():
-                self.debug(
-                    f"User '<options=bold>{new_user.preferred_username}</>' already exists in SRE '<options=bold>{self.sre_name}</>' database"
-                )
-            else:
-                self.info(
-                    f"Adding '<options=bold>{new_user.preferred_username}</>' to SRE '<options=bold>{self.sre_name}</>' database"
-                )
-                users_to_add.append(new_user)
-
-        # Add user details to the mustache template
-        pwd_date = datetime.utcnow().replace(tzinfo=timezone.utc).isoformat()
-        user_data = {
-            "users": [
-                {
-                    "username": user.user_principal_name,
-                    "email_address": user.email_address,
-                    "full_name": user.display_name,
-                    "password_hash": hex_string(64),
-                    "password_salt": hex_string(64),
-                    "password_date": pwd_date,
-                }
-                for user in users_to_add
-            ],
-            "group_name": "Research Users",
-        }
-        if user_data:
-            self.postgres_provisioner.execute_scripts(
-                [self.postgres_script_path / "add_users.mustache.sql"],
-                mustache_values=user_data,
-            )
+        raise NotImplementedError("Users should not be directly added to Guacamole.")
```

Comment:
Why leave these methods in if we will not use them?

`NotImplementedError` is probably the wrong exception here

> Note: It should not be used to indicate that an operator or method is not meant to be supported at all – in that case either leave the operator / method undefined or, if a subclass, set it to [None](https://docs.python.org/3/library/constants.html#None)

[docs](https://docs.python.org/3/library/exceptions.html#NotImplementedError)

Reply:
Fixed